types/react-country-flag/index.d.ts

```diff
@@ -0,0 +1,23 @@
+// Type definitions for react-country-flag 2.2
+// Project: https://danalloway.github.io/react-country-flag
+// Definitions by: Piotr Błażejewicz <https://github.com/peterblazejewicz>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+import * as React from 'react';
+
+// @credits @r3wt
+export interface ReactCountryFlagProps<T> extends React.DetailedHTMLProps<React.LabelHTMLAttributes<T>, T> {
+    cdnSuffix?: string;
+    /** @default 'https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.4.3/flags/4x3/'' */
+    cdnUrl?: string;
+    countryCode: string;
+    /** @default false */
+    svg?: boolean;
+    style?: React.CSSProperties;
+}
+
+/**
+ * React component for emoji/svg country flags
+ */
+declare const ReactCountryFlag: React.FC<ReactCountryFlagProps<HTMLSpanElement | HTMLImageElement>>;
+
+export default ReactCountryFlag;
```

Comment:
This is a tricky one. The package contains both CommonJS (via `"main"` in package.json) and ES Module exports (via `"module"` in package.json):

- https://unpkg.com/browse/react-country-flag@2.3.0/dist/index.js - The actual source uses `module.exports = ReactCountryFlag`. Using `export default` here would only work for someone using `--esModuleImports`. Using `export =` more accurately reflects the implementation.
- https://unpkg.com/browse/react-country-flag@2.3.0/dist/index.es.js - using `export default` correctly reflects the implementation.

TypeScript doesn't yet have full support for Node's ES Module implementation, so I would recommend switching this to `export = ReactCountryFlag`, however this comes with some added complexity. To use this definition with `export =`, you'll need to wrap `ReactCountryFlagProps` inside of a namespace also named `ReactCountryFlag` so that the props interface is still reachable via import.